Refactor checkpointImpl

anjumfatima90 opened the pull request.

anjumfatima90 committed:
Refactor checkpointImpl

tjwatson commented on dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java:
```diff
@@ -27,28 +28,36 @@
 import io.openliberty.checkpoint.spi.CheckpointHookFactory.Phase;
-import io.openliberty.checkpoint.spi.SnapshotFailed;
-import io.openliberty.checkpoint.spi.SnapshotFailed.Type;
+import io.openliberty.checkpoint.internal.CheckpointFailed;
```
imports contained in the same package as the defining class will show up as unnecessary imports.

tjwatson commented on dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java:
```diff
@@ -59,60 +68,102 @@
     CheckpointImpl(ComponentContext cc, ExecuteCRIU criu) {
         this.cc = cc;
         this.criu = criu;
+        this.checkpointFeatures = "features".equals(cc.getBundleContext().getProperty(BootstrapConstants.CHECKPOINT_PROPERTY_NAME));
```
These calls are not necessary for unit tests to set the checkpointFeatures and checkpointApplications variable.  I would think the unit tests would fail with these calls here.

tjwatson commented on dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java:
```diff
@@ -59,60 +68,102 @@
-                         @Descriptor("Directory to store the snapshot") File directory) throws SnapshotFailed {
-        doSnapshot(phase, directory);
+    @Descriptor("Do a checkpoint")
+    public void checkpoint(@Parameter(names = "-p", absentValue = "") @Descriptor("The phase to checkpoint") Phase phase,
```
We should get rid of this gogo command method altogether.  I don't think we have any need for it anymore.  The method is still used by unit tests so we can make it package private.

tjwatson commented on dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java:
```diff
@@ -27,28 +28,36 @@
                                   policyOption = ReferencePolicyOption.GREEDY),
-           property = { "osgi.command.function=snapshot", "osgi.command.scope=criu" })
-public class CheckpointImpl implements Checkpoint {
+           property = { "osgi.command.function=snapshot", "osgi.command.scope=criu", Constants.SERVICE_RANKING + ":Integer=-10000" })
```
We should get rid of the `osgi.command` service properties at this point.  There is no need to keep on maintaining this test gogo command for doing a checkpoint.

tjwatson commented on dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java:
```diff
@@ -59,60 +68,102 @@
-    @Descriptor("Take a snapshot")
-    public void snapshot(Phase phase) throws SnapshotFailed {
-        doSnapshot(phase,
+    @Descriptor("Do a checkpoint")
```
Remove the gogo `@Descriptor` annotation and make this method package private.

tjwatson commented on dev/io.openliberty.checkpoint/test/io/openliberty/checkpoint/internal/CheckpointImplTest.java:
```diff
@@ -31,8 +31,8 @@
 import io.openliberty.checkpoint.spi.CheckpointHookFactory;
-import io.openliberty.checkpoint.spi.SnapshotFailed;
-import io.openliberty.checkpoint.spi.SnapshotFailed.Type;
+import io.openliberty.checkpoint.internal.CheckpointFailed;
```
unnecessary imports.

tjwatson committed:
Fix unit tests

Unit tests do not need to check the context properties.
Intead use a constructor to set the booleans

tjwatson commented:
fixes #17754
